pkgs/http/lib/src/base_request.dart

```diff
@@ -132,13 +132,25 @@ abstract class BaseRequest {
     try {
       var response = await client.send(this);
       var stream = onDone(response.stream, client.close);
-      return StreamedResponse(ByteStream(stream), response.statusCode,
-          contentLength: response.contentLength,
-          request: response.request,
-          headers: response.headers,
-          isRedirect: response.isRedirect,
-          persistentConnection: response.persistentConnection,
-          reasonPhrase: response.reasonPhrase);
+
+      if (response is BaseResponseV2) {
+        return StreamedResponseV2(ByteStream(stream), response.statusCode,
+            contentLength: response.contentLength,
+            request: response.request,
+            headers: response.headers,
+            isRedirect: response.isRedirect,
+            url: (response as BaseResponseV2).url,
```

Comment:
I think we can avoid the cast with if-case a pattern to pull out the `url` field

```suggestion
      if (response case BaseResponseV2(:final url)) {
        return StreamedResponseV2(ByteStream(stream), response.statusCode,
            contentLength: response.contentLength,
            request: response.request,
            headers: response.headers,
            isRedirect: response.isRedirect,
            url: url,
```

Reply:
Done